firedrake/slate/slate.py

```diff
@@ -317,8 +315,8 @@ def ufl_domains(self):
         """Returns the integration domains of the integrals associated with
         the tensor.
         """
-        collected_domains = [op.ufl_domain() for op in self.operands]
-        return join_domains(collected_domains)
+        collected_domains = [op.ufl_domains() for op in self.operands]
+        return tuple(OrderedDict.fromkeys(chain(*collected_domains)))
```

Comment:
After looking in `ufl.domain`, I believe that they do the same thing...but now I realize that they are doing one additional check: `join_domains` also checks that domains with the same id are compatible. Probably should revert this back to `join_domains`